Tests/GeoLib/TestDuplicateGeometry.cpp

```diff
@@ -0,0 +1,150 @@
+			ASSERT_EQ((*(*pnts)[i])[2], (*(*new_pnts)[i])[2]);
+		}
+		mod_pnts.push_back(new GeoLib::Point(0,0,0));
+		ASSERT_EQ(mod_pnts.size(), pnts->size() + 1);
+		ASSERT_EQ(mod_pnts.size(), new_pnts->size());
+	}
+
+	// add polylines
+	std::size_t n_plys (rand() % 10 + 1);
+	std::unique_ptr<std::vector<GeoLib::Polyline*>> plys (new std::vector<GeoLib::Polyline*>);
+	for (std::size_t i=0; i<n_plys; ++i)
+	{
+		int n_ply_pnts (rand() % 100 + 2);
+		GeoLib::Polyline* line = new GeoLib::Polyline(*geo.getPointVec(input_name));
+		for (std::size_t j=0; j<n_ply_pnts; ++j)
+			line->addPoint(rand() % n_pnts);
+		plys->push_back(line);
+	}
+	geo.addPolylineVec(std::move(plys), input_name);
+
+	// duplicate polylines
+	{
+		GeoLib::DuplicateGeometry dup (geo, input_name, output);
+		std::string const& output2 = dup.getFinalizedOutputName();
+		ASSERT_FALSE(output == output2);
+
+		std::vector<GeoLib::Polyline*> const*const plys (geo.getPolylineVec(input_name));
+		std::vector<GeoLib::Polyline*> const*const new_plys (geo.getPolylineVec(output2));
+		ASSERT_EQ(n_plys, new_plys->size());
+
+		for (std::size_t i=0; i<n_plys; ++i)
+		{
+			std::size_t const n_ply_pnts ((*new_plys)[i]->getNumberOfPoints());
+			ASSERT_EQ(n_ply_pnts, (*plys)[i]->getNumberOfPoints());
+			ASSERT_EQ((*new_plys)[i]->getNumberOfSegments(), (*plys)[i]->getNumberOfSegments());
+			for (std::size_t j=0; j<n_ply_pnts; ++j)
+				ASSERT_EQ((*plys)[i]->getPointID(j), (*new_plys)[i]->getPointID(j));
+		}
+
+		std::vector<GeoLib::Polyline*>& mod_plys (dup.getPolylineVectorCopy());
+		mod_plys.push_back(new GeoLib::Polyline(*(*new_plys)[0]));
+		ASSERT_EQ(mod_plys.size(), plys->size() + 1);
+		ASSERT_EQ(mod_plys.size(), new_plys->size());
+	}
+
+	// add surfaces
+	std::size_t n_sfcs (rand() % 10 + 1);
+	std::unique_ptr<std::vector<GeoLib::Surface*>> sfcs (new std::vector<GeoLib::Surface*>);
+	for (std::size_t i=0; i<n_sfcs; ++i)
+	{
+		int n_tris (rand() % 10);
+		GeoLib::Surface* sfc = new GeoLib::Surface(*geo.getPointVec(input_name));
+		for (std::size_t j=0; j<n_tris; ++j)
+			sfc->addTriangle(rand() % n_pnts, rand() % n_pnts, rand() % n_pnts);
+		if (sfc->getNTriangles() > 0)
+			sfcs->push_back(sfc);
+		else
+			delete sfc;
+	}
+	n_sfcs = sfcs->size();
+	geo.addSurfaceVec(std::move(sfcs), input_name);
+
+	// duplicate surfaces
+	{
+		GeoLib::DuplicateGeometry dup (geo, input_name, output);
+		std::string const& output2 = dup.getFinalizedOutputName();
+		ASSERT_FALSE(output == output2);
+
+		std::vector<GeoLib::Surface*> const* sfcs (geo.getSurfaceVec(input_name));
+		std::vector<GeoLib::Surface*> const* new_sfcs (geo.getSurfaceVec(output2));
+		ASSERT_EQ(n_sfcs, new_sfcs->size());
+
+		for (std::size_t i=0; i<n_sfcs; ++i)
+		{
+			std::size_t const n_tris ((*new_sfcs)[i]->getNTriangles());
+			ASSERT_EQ(n_tris, (*sfcs)[i]->getNTriangles());
+			for (std::size_t j=0; j<n_tris; ++j)
+				for (std::size_t k=0; k<3; ++k)
+					ASSERT_EQ((*(*(*sfcs)[i])[j])[k], (*(*(*new_sfcs)[i])[j])[k]);
```

Comment:
An == op on this level doesn't make sense except for theoretical test cases. Surfaces should be evalutated as identical if they are topological identical. This includes variances in id's, triangle sequences or even slight differences in elevation because for all practical purposes these values will vary due to floating point inaccuracies, mesh algorithms and the like.

I agree that the above line doesn't look nice but it gets the job done and it's just for a test.